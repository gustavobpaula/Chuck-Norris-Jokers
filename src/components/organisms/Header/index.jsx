import React from 'react';
import { Logo } from '../../molecules';

const Header = () => <Logo />;

export default Header;
