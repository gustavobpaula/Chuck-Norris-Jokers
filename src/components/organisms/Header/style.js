import styled from 'styled-components';
import designSystem from '../../../designSystem';

export const Header = styled.header`
  background-color: ${designSystem.get('colors.color1')};
  min-height: 100px;
  padding: 20px 0 0 0;
  text-align: center;
  width: 100%;
`;
