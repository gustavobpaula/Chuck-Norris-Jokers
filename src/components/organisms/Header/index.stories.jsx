import React from 'react';
import { BrowserRouter } from 'react-router-dom';
import Header from './index';

export default {
  component: Header,
  title: 'Organisms/Header',
};

export const element = () => (
  <BrowserRouter>
    <Header />
  </BrowserRouter>
);

element.story = {
  name: 'Default',
};
