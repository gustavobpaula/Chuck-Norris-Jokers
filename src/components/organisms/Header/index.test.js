import React from 'react';
import { mount } from 'enzyme';
import { BrowserRouter } from 'react-router-dom';
import Header from './index';

describe('<Header />', () => {
  it('should mount properly', async () => {
    const wrapper = mount(
      <BrowserRouter>
        <Header />
      </BrowserRouter>
    );

    expect(wrapper).toMatchSnapshot();
  });
});
