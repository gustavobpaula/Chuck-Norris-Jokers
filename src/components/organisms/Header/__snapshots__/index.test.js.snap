// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Header /> should mount properly 1`] = `
<BrowserRouter>
  <Router
    history={
      Object {
        "action": "POP",
        "block": [Function],
        "createHref": [Function],
        "go": [Function],
        "goBack": [Function],
        "goForward": [Function],
        "length": 1,
        "listen": [Function],
        "location": Object {
          "hash": "",
          "pathname": "/",
          "search": "",
          "state": undefined,
        },
        "push": [Function],
        "replace": [Function],
      }
    }
  >
    <Header>
      <style__Header>
        <StyledComponent
          forwardedComponent={
            Object {
              "$$typeof": Symbol(react.forward_ref),
              "attrs": Array [],
              "componentStyle": ComponentStyle {
                "componentId": "style__Header-sc-1ve0xpd-0",
                "isStatic": false,
                "lastClassName": "hYlzqU",
                "rules": Array [
                  "background-color:",
                  "#e8e8e8",
                  ";min-height:100px;padding:20px 0 0 0;text-align:center;width:100%;",
                ],
              },
              "displayName": "style__Header",
              "foldedComponentIds": Array [],
              "render": [Function],
              "styledComponentId": "style__Header-sc-1ve0xpd-0",
              "target": "header",
              "toString": [Function],
              "warnTooManyClasses": [Function],
              "withComponent": [Function],
            }
          }
          forwardedRef={null}
        >
          <header
            className="style__Header-sc-1ve0xpd-0 hYlzqU"
          >
            <Logo
              className="logo"
              height="auto"
              maxHeight="100px"
              maxWidth="100%"
              width="auto"
            >
              <style__Link
                className="logo"
                title="Home"
                to="/"
              >
                <StyledComponent
                  className="logo"
                  forwardedComponent={
                    Object {
                      "$$typeof": Symbol(react.forward_ref),
                      "attrs": Array [],
                      "componentStyle": ComponentStyle {
                        "componentId": "style__Link-sc-8pdh8h-0",
                        "isStatic": false,
                        "lastClassName": "hxIPvP",
                        "rules": Array [
                          "align-items:center;color:",
                          "#00334e",
                          ";display:flex;justify-content:center;text-decoration:none;",
                        ],
                      },
                      "displayName": "style__Link",
                      "foldedComponentIds": Array [],
                      "render": [Function],
                      "styledComponentId": "style__Link-sc-8pdh8h-0",
                      "target": [Function],
                      "toString": [Function],
                      "warnTooManyClasses": [Function],
                      "withComponent": [Function],
                    }
                  }
                  forwardedRef={null}
                  title="Home"
                  to="/"
                >
                  <LinkTo
                    className="style__Link-sc-8pdh8h-0 hxIPvP logo"
                    title="Home"
                    to="/"
                  >
                    <Link
                      className="style__Link-sc-8pdh8h-0 hxIPvP logo"
                      title="Home"
                      to="/"
                    >
                      <LinkAnchor
                        className="style__Link-sc-8pdh8h-0 hxIPvP logo"
                        href="/"
                        navigate={[Function]}
                        title="Home"
                      >
                        <a
                          className="style__Link-sc-8pdh8h-0 hxIPvP logo"
                          href="/"
                          onClick={[Function]}
                          title="Home"
                        >
                          <Image
                            alt="Chuck Norris Logo"
                            height="auto"
                            maxHeight="100px"
                            maxWidth="100%"
                            src="test-file-stub"
                            width="auto"
                          >
                            <style__Image
                              alt="Chuck Norris Logo"
                              height="auto"
                              maxHeight="100px"
                              maxWidth="100%"
                              src="test-file-stub"
                              width="auto"
                            >
                              <StyledComponent
                                alt="Chuck Norris Logo"
                                forwardedComponent={
                                  Object {
                                    "$$typeof": Symbol(react.forward_ref),
                                    "attrs": Array [],
                                    "componentStyle": ComponentStyle {
                                      "componentId": "style__Image-zsbil-0",
                                      "isStatic": false,
                                      "lastClassName": "nQWWf",
                                      "rules": Array [
                                        "height:",
                                        [Function],
                                        ";max-height:",
                                        [Function],
                                        ";max-width:",
                                        [Function],
                                        ";width:",
                                        [Function],
                                        ";",
                                      ],
                                    },
                                    "displayName": "style__Image",
                                    "foldedComponentIds": Array [],
                                    "render": [Function],
                                    "styledComponentId": "style__Image-zsbil-0",
                                    "target": "img",
                                    "toString": [Function],
                                    "warnTooManyClasses": [Function],
                                    "withComponent": [Function],
                                  }
                                }
                                forwardedRef={null}
                                height="auto"
                                maxHeight="100px"
                                maxWidth="100%"
                                src="test-file-stub"
                                width="auto"
                              >
                                <img
                                  alt="Chuck Norris Logo"
                                  className="style__Image-zsbil-0 nQWWf"
                                  height="auto"
                                  src="test-file-stub"
                                  width="auto"
                                />
                              </StyledComponent>
                            </style__Image>
                          </Image>
                          <Typography
                            component="span"
                            variant="logo"
                          >
                            <style__Typography
                              as="span"
                              variant="logo"
                            >
                              <StyledComponent
                                as="span"
                                forwardedComponent={
                                  Object {
                                    "$$typeof": Symbol(react.forward_ref),
                                    "attrs": Array [],
                                    "componentStyle": ComponentStyle {
                                      "componentId": "style__Typography-sc-8vyb1l-0",
                                      "isStatic": false,
                                      "lastClassName": "jIxkTh",
                                      "rules": Array [
                                        [Function],
                                      ],
                                    },
                                    "displayName": "style__Typography",
                                    "foldedComponentIds": Array [],
                                    "render": [Function],
                                    "styledComponentId": "style__Typography-sc-8vyb1l-0",
                                    "target": "span",
                                    "toString": [Function],
                                    "warnTooManyClasses": [Function],
                                    "withComponent": [Function],
                                  }
                                }
                                forwardedRef={null}
                                variant="logo"
                              >
                                <span
                                  className="style__Typography-sc-8vyb1l-0 jIxkTh"
                                >
                                  Chuck "Joke" Norris
                                </span>
                              </StyledComponent>
                            </style__Typography>
                          </Typography>
                        </a>
                      </LinkAnchor>
                    </Link>
                  </LinkTo>
                </StyledComponent>
              </style__Link>
            </Logo>
          </header>
        </StyledComponent>
      </style__Header>
    </Header>
  </Router>
</BrowserRouter>
`;
